azurerm/internal/services/eventgrid/tests/eventgrid_event_subscription_resource_test.go

```diff
@@ -401,3 +424,52 @@ resource "azurerm_eventgrid_event_subscription" "test" {
 }
 `, data.RandomInteger, data.Locations.Primary, data.RandomString, data.RandomInteger, data.RandomInteger)
 }
+
+func testAccAzureRMEventGridEventSubscription_advancedFilter(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+resource "azurerm_storage_account" "test" {
+  name                     = "acctestacc%s"
+  resource_group_name      = "${azurerm_resource_group.test.name}"
+  location                 = "${azurerm_resource_group.test.location}"
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+  tags = {
+    environment = "staging"
+  }
+}
+resource "azurerm_storage_queue" "test" {
+  name                 = "mysamplequeue-%d"
+  resource_group_name  = "${azurerm_resource_group.test.name}"
+  storage_account_name = "${azurerm_storage_account.test.name}"
+}
+resource "azurerm_eventgrid_event_subscription" "test" {
+  name  = "acctesteg-%d"
+  scope = "${azurerm_storage_account.test.id}"
+  storage_queue_endpoint {
+    storage_account_id = "${azurerm_storage_account.test.id}"
+    queue_name         = "${azurerm_storage_queue.test.name}"
+  }
+  advanced_filter {
+    number_less_than {
+      key   = "data.contentLength"
+      value = 42.0
+    }
+    number_in {
+      key    = "data.contentLength"
+      values = [1, 1, 2, 3, 5]
```

Comment:
is that an intentional duplicate value?

Reply:
No, other than that it works with the API, there is no deeper meaning in it. Was just part of the Fibonacci sequence.

Reply:
ok, seems reasonable to me :)